regression/apps/include/test.h

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MPL-2.0
+
+/*
+ * A framework for writing regression tests.
```

Comment:
This framework looks quite interesting. The test cases based on this framework, as shown in `tcp_err` and `udp_err`, look neat. My major concern is about the generality of this framework. It promotes the test author to organize his test cases in a very opinionated fashion. I don't know if this fashion is equally good to writing non-network test cases. And if other test authors would like to adopt it. So I suggest moving `include/test.h` to `network/`.

Reply:
> So I suggest moving include/test.h to network/.

That's my first thought. I have no real objection to this idea, because if we find it useful later, we can easily move it out.

But here is another example that uses this framework to check the correctness of file I/O system calls: #711